### PR TITLE
1890: Create manual unit tests for Format Preview

### PR DESCRIPTION
**Pull request by chrismiceli**
proof of concept for automated javascript testing

### CODE REVIEW

**Comment by chrismiceli** [comment]
@70ray, this sets up qunit and adds a node script to run it through travis with firefox headless using selenium. If this looks good, we can expand the actual tests from the dummy one I have right now.

File: package.json [hunk 1]
```diff
@@ -22,6 +23,8 @@
   },
   "homepage": "https://github.com/DistributedProofreaders/dproofreaders",
   "devDependencies": {
-    "eslint": "^6.8.0"
+    "eslint": "^7.1.0",
```
Comment (chrismiceli) [inline package.json:26]: Also took the opportunity to update eslint to the latest version.

**Comment by chrismiceli** [comment]
You can see a travis build on my fork here: https://travis-ci.org/github/chrismiceli/dproofreaders/jobs/691583500
image

File: .travis.yml [hunk 2]
```diff
@@ -32,6 +34,7 @@ script:
     # linting is SUCH a low bar, but it is a bar!
     - ./SETUP/lint_php_files.sh
     - npm run lint
+    - npm run test
```
Comment (cpeel) [inline .travis.yml:37]: Move this to below the `phpunit` tests below, since they're tests not linting.

File: jsUnit.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/* global process require */
```
Comment (cpeel) [inline jsUnit.js:1]: Let's move this file to `SETUP/tests` too so it isn't hanging around in deployed site code.

**Comment by 70ray** [comment]
> @70ray, this sets up qunit and adds a node script to run it through travis with firefox headless using selenium. If this looks good, we can expand the actual tests from the dummy one I have right now.

I've made a modified version of preview.js with an 'analyse' function and some tests for it which I've tested locally using qunit.html. When your branch is merged I can put them in.

**Comment by chrismiceli** [comment]
Here is what it looks like when you manually execute the tests by visiting the qunit.html page:
image

**Comment by cpeel** [comment]
Thanks @chrismiceli! Can you rebase this to resolve the merge conflicts? I'll get this merged in and then tweak some of the `SETUP/tests` contents (move the phpunit tests into a subdirectory, update the `README.md`s, etc).

**Comment by chrismiceli** [comment]
> Thanks @chrismiceli! Can you rebase this to resolve the merge conflicts? I'll get this merged in and then tweak some of the `SETUP/tests` contents (move the phpunit tests into a subdirectory, update the `README.md`s, etc).

Ok, all merge conflicts should be resolved.

**Comment by cpeel** [comment]
Github is still showing that there are conflicts :/

**Comment by chrismiceli** [comment]
> Github is still showing that there are conflicts :/

Weird, it doesn't show anything for me and I was able to merge into my local master branch with no issues. What files shows conflict?

**Comment by cpeel** [comment]
Sadly, it doesn't give me the details. I will get this merged in locally off of your branch, standby.

**Pull request merged**